config/configuration.go

```diff
@@ -27,24 +27,35 @@ import (
 
 var log = logging.Logger("config")
 
-// configMu protects the config from read/write
-var configMu sync.RWMutex
-
-// config holds the current node config
-var config *Configuration
-
-// Config returns the current loaded config
-func Config() *Configuration {
-	configMu.RLock()
-	defer configMu.RUnlock()
-	return config
-}
-
-// SetConfig sets the config
-func SetConfig(c *Configuration) {
-	configMu.Lock()
-	defer configMu.Unlock()
-	config = c
+type Config interface {
```

Comment:
Added config interface so it can be mocked during tests. In addition for multi-tenant nodes, will come handy when we split up in nodeConfig and tenantConfig.